pkg/apis/v1alpha1/ecsnodeclass.go

```diff
@@ -212,15 +213,24 @@ type SystemDisk struct {
 	// +kubebuilder:default:={"cloud","cloud_efficiency","cloud_ssd","cloud_essd","cloud_auto","cloud_essd_entry"}
 	// +optional
 	Categories []string `json:"categories,omitempty"`
-	// The size of the system disk. Unit: GiB.
-	// Valid values:
-	//   * If you set Category to cloud: 20 to 500.
-	//   * If you set Category to other disk categories: 20 to 2048.
+	// Size in `Gi`, `G`, `Ti`, or `T`. You must specify either a snapshot ID or
+	// a volume size. The following are the supported volumes sizes for each volume
+	// type:
+	//
+	//    * gp2 and gp3: 1-16,384
+	//
+	//    * io1 and io2: 4-16,384
+	//
+	//    * st1 and sc1: 125-16,384
 	//
-	// +kubebuilder:validation:XValidation:message="size invalid",rule="self >= 20"
-	// +kubebuilder:default:=20
+	//    * standard: 1-1,024
+	// + TODO: Add the CEL resources.quantity type after k8s 1.29
+	// + https://github.com/kubernetes/apiserver/commit/b137c256373aec1c5d5810afbabb8932a19ecd2a#diff-838176caa5882465c9d6061febd456397a3e2b40fb423ed36f0cabb1847ecb4dR190
+	// +kubebuilder:validation:Pattern:="^((?:[1-9][0-9]{0,3}|[1-4][0-9]{4}|[5][0-8][0-9]{3}|59000)Gi|(?:[1-9][0-9]{0,3}|[1-5][0-9]{4}|[6][0-3][0-9]{3}|64000)G|([1-9]||[1-5][0-7]|58)Ti|([1-9]||[1-5][0-9]|6[0-3]|64)T)$"
+	// +kubebuilder:validation:Schemaless
+	// +kubebuilder:validation:Type:=string
 	// +optional
-	Size *int32 `json:"size,omitempty"`
+	Size *resource.Quantity `json:"size,omitempty" hash:"string"`
```

Comment:
I think we should introduce a new field, like VolumeSize

If we do not do like this, the upgrading will meet problem.

Reply:
I updated the description and thought maybe adding motivation could help improve understanding.